src/Microsoft.Extensions.Configuration.Abstractions/IConfigurationSource.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Represents a source of configuration key/values for an application.
+    /// </summary>
+    public interface IConfigurationSource
+    {
+        /// <summary>
+        /// Builds the <see cref="IConfigurationProvider"/> for this source.
+        /// </summary>
+        /// <param name="builder">The <see cref="IConfigurationBuilder"/>.</param>
+        /// <returns></returns>
+        IConfigurationProvider Build(IConfigurationBuilder builder);
```

Comment:
If we pass the builder we can use the extension methods to get the FileProvider, otherwise we have to hardcode constants for the keys